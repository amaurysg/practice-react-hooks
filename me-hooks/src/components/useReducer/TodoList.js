import React from "react";
import TodoListItem from "./TodoListItem";

const TodoList = ({ todos, handleDelete, handleToogle, handleUpdate }) => {
  return (
    <>
      <ul className="list-group">
        {todos.map((todo, i) => {
          return (
            // TodoListItem , todo, index, funciones delete, toogle
            <TodoListItem
              key={todo.id}
              todo={todo}
              i={i}
              handleDelete={handleDelete}
              handleToogle={handleToogle}
              handleUpdate={handleUpdate}
            />
          );
        })}
      </ul>
    </>
  );
};

export default TodoList;
