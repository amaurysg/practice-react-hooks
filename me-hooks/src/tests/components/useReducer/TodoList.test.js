import { shallow } from "enzyme";
import TodoList from "../../../components/useReducer/TodoList";
import demoTodos from "../../fixtures/demoTodos";

describe("Test in component <TodoList />", () => {
  const handleDelete = jest.fn();
  const handleToogle = jest.fn();
  const handleUpdate = jest.fn();

  const wrapper = shallow(
    <TodoList
      todos={demoTodos}
      handleDelete={handleDelete}
      handleToogle={handleToogle}
      handleUpdate={handleUpdate}
    />
  );

  test(" Debe renderizar correctamente ", () => {
    expect(wrapper).toMatchSnapshot();
  });
  test(" Debe tener dos <TodoListItem/> ", () => {
    expect(wrapper.find("TodoListItem").length).toBe(demoTodos.length);
    expect(wrapper.find("TodoListItem").at(0).prop("handleDelete")).toEqual(
      expect.any(Function)
    );
  });
});
