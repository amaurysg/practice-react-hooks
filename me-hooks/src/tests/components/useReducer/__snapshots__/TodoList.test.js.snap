// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in component <TodoList />  Debe renderizar correctamente  1`] = `
<Fragment>
  <ul
    className="list-group"
  >
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToogle={[MockFunction]}
      handleUpdate={[MockFunction]}
      i={0}
      key="1"
      todo={
        Object {
          "desc": "React",
          "done": false,
          "id": 1,
        }
      }
    />
    <TodoListItem
      handleDelete={[MockFunction]}
      handleToogle={[MockFunction]}
      handleUpdate={[MockFunction]}
      i={1}
      key="2"
      todo={
        Object {
          "desc": "Jest",
          "done": false,
          "id": 2,
        }
      }
    />
  </ul>
</Fragment>
`;
